apis/v1alpha1/gateway_types.go

```diff
@@ -330,6 +359,17 @@ type GatewayTLSConfig struct {
 	//
 	// +optional
 	CertificateRef CertificateObjectReference `json:"certificateRef,omitempty"`
+
+	// AllowRouteOverride dictates if TLS settings can be configured
+	// via Routes or not.
+	//
+	// CertificateRef must be defined even if allowRouteOverride.certificate is
+	// set to 'AllowAll' as it will be used as the default certificate for the
+	// listener.
+	//
+	// +optional
+	AllowRouteOverride TLSOverridePolicy `json:"allowRouteOverride"`
```

Comment:
Hmm we should think about a better name:

```
AllowRouteOverride: Allow
```

how about?

```
RoutePolicy: AllowOverride

or

RouteOverride: Allow
```

Reply:
I'm in favor of either of ^ that @bowei suggests.

Reply:
Since this would be a per-attribute struct, I think I prefer `RouteOverride` as the name here. That would translate into something like this:

```
routeOverride:
  certificate: Allow
```

And in the future that might be expanded with more fields and/or more potential values such as `AllowNamespaceList`.